src/crawlers/browser_crawler.js

```diff
@@ -351,6 +351,19 @@ export default class BrowserCrawler extends BasicCrawler {
         });
     }
 
+    /**
+     * Returns context options that will be used when a page needs to be created.
+     * Required for proxy per page.
+     *
+     * @param {BrowserCrawlingContext & CrawlingContext} crawlingContext
+     * @ignore
+     * @protected
+     * @internal
+     */
+    _getNewPageOptions() {
+        return undefined;
```

Comment:
No need to keep it empty. We can return the `id` and add the MITM handling. See the comments below.

Reply:
How would it come useful?